File: core/format_rules.py
from __future__ import annotations

import re

PLATFORMS = ("Reddit", "LinkedIn", "Other")


def apply_format_rules(text: str, platform: str) -> str:
    """Mechanical, platform-specific formatting only — never touches voice."""
    text = text.strip()

    if platform == "Reddit":
        return _format_reddit(text)
    if platform == "LinkedIn":
        return _format_linkedin(text)
    return text


def _format_reddit(text: str) -> str:
    # Strip any hashtags — not a Reddit convention.
    text = re.sub(r"(?<!\S)#\w+", "", text)
    text = re.sub(r"[ \t]+\n", "\n", text)
    text = re.sub(r"\n{3,}", "\n\n", text)
    return text.strip()


def _format_linkedin(text: str) -> str:
    # LinkedIn favors short paragraphs with a blank line between them.
    paragraphs = [p.strip() for p in text.split("\n\n") if p.strip()]
    return "\n\n".join(paragraphs)

File: macos-app/Sources/HumanizerApp/FormatRules.swift
import Foundation

enum FormatRules {
    static let platforms = ["Reddit", "LinkedIn", "Other"]

    /// Mechanical, platform-specific formatting only — never touches voice.
    static func apply(_ text: String, platform: String) -> String {
        let trimmed = text.trimmingCharacters(in: .whitespacesAndNewlines)
        switch platform {
        case "Reddit": return formatReddit(trimmed)
        case "LinkedIn": return formatLinkedIn(trimmed)
        default: return trimmed
        }
    }

    private static func formatReddit(_ text: String) -> String {
        var result = text
        result = result.replacingOccurrences(
            of: #"(?<!\S)#\w+"#, with: "", options: .regularExpression
        )
        result = result.replacingOccurrences(
            of: #"[ \t]+\n"#, with: "\n", options: .regularExpression
        )
        result = result.replacingOccurrences(
            of: #"\n{3,}"#, with: "\n\n", options: .regularExpression
        )
        return result.trimmingCharacters(in: .whitespacesAndNewlines)
    }

    private static func formatLinkedIn(_ text: String) -> String {
        let paragraphs = text.components(separatedBy: "\n\n")
            .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
            .filter { !$0.isEmpty }
        return paragraphs.joined(separator: "\n\n")
    }
}
